### DISCUSSION TITLE
osu! Lazer shaking cursor

### DISCUSSION THREAD

**Discussion by chicocaine** [ppy/osu#24469] [Q&A] [answered]
\I've been playing osu! Lazer for a couple of weeks recently. We all know that osu! Lazer has an integrated tablet driver that supports a wide range of tablets which includes, the wacom ctl-472 (the one I'm using). In osu! Lazer, I can't drag while using the wacom drivers in-game because the cursor will stick into a corner of the screen. Essentially, you're compelled to use the integrated drivers to drag. Now this isn't that much of a problem, I got used to it pretty quick. However, when I use the integrated drivers, the cursor shakes fast (say the amplitude is not big, but the frequency is). You don't notice it while playing the maps since the cursor is always moving but on short and long breaks, even in the menu, I just get slightly annoyed by it. I don't know if it's the engine, the driver, my tablet, or how to fix it, hence this post. Anyone know anything about this?

**Comment by n0Oo0Oo0b** [comment, marked as answer]
From my rough testing, it seems like the tablet is causing the jitter since the it happens even when the pen is staitionary (CTL-472, happens on both integrated driver and OpenTabletDriver without smoothing). Wacom's drivers apply smoothing by default, so that would explain why the jitter doesn't happen using those. A possible solution could be to add customizable smoothing to lazer's drivers (or maybe even custom filters) since it's pretty commonplace in stable. For now, you could use OpenTabletDriver with AbstractQbit's smoothing plugin instead.

**Reply by chicocaine** [reply]
That makes sense. Thanks.

**Reply by n0Oo0Oo0b** [reply]
> A possible solution could be to add customizable smoothing to lazer's drivers (or maybe even custom filters)

#15306
